File: src/Contexts/Rutine/Exercises/domain/ExerciseName.ts
import { InvalidArgumentError } from '../../../Shared/domain/value-object/InvalidArgumentError';
import { StringValueObject } from '../../../Shared/domain/value-object/StringValueObject';

export class ExerciseName extends StringValueObject {
  constructor(value: string) {
    super(value);
    this.ensureLengthIsLessThan50Characters(value);
  }

  private ensureLengthIsLessThan50Characters(value: string): void {
    if (value.length > 50) {
      throw new InvalidArgumentError(`The Exercise Name <${value}> has more than 50 characters`);
    }
  }
}

File: src/main/java/com/kuro/rutineweb/exercises/entities/ExerciseName.java
package com.kuro.rutineweb.exercises.entities;

import com.kuro.rutineweb.Shared.entities.valueObject.InvalidArgumentError;
import com.kuro.rutineweb.Shared.entities.valueObject.StringValueObject;

public class ExerciseName extends StringValueObject {
    public ExerciseName(String value) {
        super(value);
        ensureLengthIsLessThan50Characters(value);
    }

    private void ensureLengthIsLessThan50Characters(String value) {
        if(value.length() > 50) {
            throw new InvalidArgumentError("The Exercise Name " + value + " has more than 50 characters");
        }
    }
}
